src/mdbExtensionController.ts

```diff
@@ -310,30 +311,34 @@ export default class MDBExtensionController implements vscode.Disposable {
     );
     this.registerCommand(
       EXTENSION_COMMANDS.CONNECT_WITH_PARTICIPANT,
-      (id?: string) =>
-        this._participantController.connectWithParticipant(
-          id ? decodeURIComponent(id) : id
-        )
+      (data: any) => {
+        return this._participantController.connectWithParticipant(data);
+      }
     );
     this.registerCommand(
       EXTENSION_COMMANDS.SELECT_DATABASE_WITH_PARTICIPANT,
-      (_data: string) => {
-        const data = JSON.parse(decodeURIComponent(_data));
-        return this._participantController.selectDatabaseWithParticipant({
-          chatId: data.chatId,
-          databaseName: data.databaseName,
-        });
+      (data: any) => {
+        return this._participantController.selectDatabaseWithParticipant(data);
       }
     );
     this.registerCommand(
       EXTENSION_COMMANDS.SELECT_COLLECTION_WITH_PARTICIPANT,
-      (_data: string) => {
-        const data = JSON.parse(decodeURIComponent(_data));
-        return this._participantController.selectCollectionWithParticipant({
-          chatId: data.chatId,
-          databaseName: data.databaseName,
-          collectionName: data.collectionName,
+      (data: any) => {
+        return this._participantController.selectCollectionWithParticipant(
+          data
+        );
+      }
+    );
+    this.registerCommand(
+      EXTENSION_COMMANDS.PARTICIPANT_OPEN_RAW_SCHEMA_OUTPUT,
+      async ({ schema }: OpenSchemaCommandArgs) => {
+        const document = await vscode.workspace.openTextDocument({
+          language: 'json',
+          content: schema,
         });
+        await vscode.window.showTextDocument(document, { preview: true });
```

Comment:
Should we show the schema in the Playground Result pane, like we display the result of executable code blocks from chat participants?

Reply:
I went with this approach, using a new json document view with preview, instead of a playground result pane for a couple reasons:
- It allows for JSON support.
- A preview is more ephemeral which is what I was thinking we want for something like this that folks want to copy paste or view quickly. It's replaced by other previews as well.
- This is the result of a schema run, and not something that is from running a playground, so naming it wouldn't be quite right.
Are there reasons we would want to reuse the result pane?

Reply:
In the results panel, you can also set the json language. I don't have a strong opinion on where the schema output should be, more like raising a question to think about. For me, the schema output looks pretty much the same as the export to language output, which we also show in the results pane. Only Shema will behave differently if we open it as a preview text document. On the other hand, we open documents from the sidebar also as json pages. If you feel strongly about this behavior, I don't mind keeping it as is. Just want to make sure this is what users would feel more natural about.